include/alpaka/acc/AccGpuHipRt.hpp

```diff
@@ -0,0 +1,312 @@
+/**
+* \file
+* Copyright 2014-2016 Benjamin Worpitz, Rene Widera
+*
+* This file is part of alpaka.
+*
+* alpaka is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* alpaka is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with alpaka.
+* If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#ifdef ALPAKA_ACC_GPU_HIP_ENABLED
+
+#include <alpaka/core/Common.hpp>                   // ALPAKA_FN_*, BOOST_LANG_HIP
+
+// Base classes.
+
+#include <alpaka/workdiv/WorkDivHipBuiltIn.hpp>
+#include <alpaka/idx/gb/IdxGbHipBuiltIn.hpp>       // IdxGbHipBuiltIn
+#include <alpaka/idx/bt/IdxBtHipBuiltIn.hpp>	   // IdxBtHipBuiltIn
+#include <alpaka/atomic/AtomicHipBuiltIn.hpp>
+#include <alpaka/atomic/AtomicHierarchy.hpp>    // AtomicHierarchy
+#include <alpaka/math/MathHipBuiltIn.hpp>
+#include <alpaka/block/shared/dyn/BlockSharedMemDynHipBuiltIn.hpp>
+#include <alpaka/block/shared/st/BlockSharedMemStHipBuiltIn.hpp>
+#include <alpaka/block/sync/BlockSyncHipBuiltIn.hpp>
+#include <alpaka/rand/RandHipRand.hpp>
+#include <alpaka/time/TimeHipBuiltIn.hpp>
+
+// Specialized traits.
+#include <alpaka/acc/Traits.hpp>                    // acc::traits::AccType
+#include <alpaka/dev/Traits.hpp>                    // dev::traits::DevType
+#include <alpaka/exec/Traits.hpp>                   // exec::traits::ExecType
+#include <alpaka/pltf/Traits.hpp>                   // pltf::traits::PltfType
+#include <alpaka/idx/Traits.hpp>
+
+// Implementation details.
+#include <alpaka/dev/DevHipRt.hpp>	// DevHipRt
+#include <alpaka/core/Hip.hpp>
+
+#include <boost/predef.h>                           // workarounds
+
+#include <typeinfo>                                 // typeid
+
+namespace alpaka
+{
+    namespace exec
+    {
+        template<
+            typename TDim,
+            typename TIdx,
+            typename TKernelFnObj,
+            typename... TArgs>
+        class ExecGpuHipRt;
+    }
+    namespace acc
+    {
+        //#############################################################################
+        //! The GPU HIP accelerator.
+        //!
+        //! This accelerator allows parallel kernel execution on devices supporting HIP or HCC
+        //#############################################################################
```

Comment:
Please remove the second `//################` and `//--------------` lines everywhere because they have been removed in the rest of the code.